### PR TITLE
All announcements having the status "deleted" must disappear definitively

### PR DESCRIPTION
**Pull request by mmarchois**
PR 20

See https://github.com/poledev/mathieu/issues/15

### CODE REVIEW

File: src/SensioLabs/JobBoardBundle/Listener/JobListener.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * @param LifecycleEventArgs $args
+     */
+    public function preUpdate(LifecycleEventArgs $args)
+    {
+        $job = $args->getEntity();
+
+        if ($job instanceof Job && null !== $this->loadedJob) {
+            if (Job::STATUS_DELETED === $job->getStatus() && $this->loadedJob->getStatus() !== Job::STATUS_DELETED) {
+                $job->setDeletedAt(new \DateTime());
```
Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Listener/JobListener.php]: https://github.com/l3pp4rd/DoctrineExtensions/blob/master/doc/timestampable.md#using-dependency-of-property-changes

File: src/SensioLabs/JobBoardBundle/Command/JobCommand.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        ;
+    }
+
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return void
+     */
+    public function execute(InputInterface $input, OutputInterface $output)
```
Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Command/JobCommand.php]: Do you know the new [SymfonyStyle](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Console/Style/SymfonyStyle.php) class ? https://github.com/symfony/symfony/pull/12035
:D 


--> [ ]

_(Seriously, this is not required at all. But you can give it a try if you have the time. Having a more conventional outputs for command line (+the helps this class give to build easily those outputs) is a great thing IMO.)_

Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Command/JobCommand.php]: I can try yes

File: src/SensioLabs/JobBoardBundle/Entity/Job.php [hunk 3]
```diff
@@ -171,6 +171,14 @@ class Job implements RoutedItemInterface
      */
     protected $publishedEnd;
 
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type = "datetime", name = "deleted_at", nullable = true)
```
Comment (florianbelhomme) [inline src/SensioLabs/JobBoardBundle/Entity/Job.php]: les espaces sont en trop autour des =

File: src/SensioLabs/JobBoardBundle/Repository/JobRepository.php [hunk 4]
```diff
@@ -164,4 +164,19 @@ private function setPublicationDateExpr(QueryBuilder $qb)
...
+
+    public function findJobsToRemove()
+    {
+        $qb = $this->createQueryBuilder('j');
+
+        $qb
+            ->delete()
+            ->where('j.status = :status')
+            ->setParameter('status', Job::STATUS_DELETED)
+            ->andWhere('j.deletedAt <= :date')
```
Comment (florianbelhomme) [inline src/SensioLabs/JobBoardBundle/Repository/JobRepository.php]: <= ou juste < ?

Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Repository/JobRepository.php]: <= pour avoir les dates du même jour + timestamp

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 5]
```diff
@@ -91,7 +91,7 @@ public function updateAction(Request $request, Job $job)
      *      name = "job_remove"
      * )
      * @ParamConverter("job", class="SensioLabsJobBoardBundle:Job")
-     * @Method({"GET"})
+     * @Method({"DELETE"})
```
Comment (florianbelhomme) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: Que se passe-t-il si quelqu'un fait un script JS qui appelle les urls "remove" en boucle ?

Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: Je vais rajouter la protection par csrf token.

Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: Not in this PR ?

**Pull request merged**